plugins/inputs/jolokia2_agent/jolokia2_agent_test.go

```diff
@@ -782,31 +773,16 @@ func SetupPlugin(t *testing.T, conf string) telegraf.Input {
 
 	for name := range table.Fields {
 		object := table.Fields[name]
-		switch name {
-		case "jolokia2_agent":
-			plugin := jolokia2_agent.JolokiaAgent{
-				Metrics:               []common.MetricConfig{},
-				DefaultFieldSeparator: ".",
-			}
-
-			if err := toml.UnmarshalTable(object.(*ast.Table), &plugin); err != nil {
-				t.Fatalf("Unable to parse jolokia_agent plugin config! %v", err)
-			}
-
-			return &plugin
-
-		case "jolokia2_proxy":
-			plugin := jolokia2_proxy.JolokiaProxy{
-				Metrics:               []common.MetricConfig{},
-				DefaultFieldSeparator: ".",
-			}
-
-			if err := toml.UnmarshalTable(object.(*ast.Table), &plugin); err != nil {
-				t.Fatalf("Unable to parse jolokia_proxy plugin config! %v", err)
-			}
+		plugin := jolokia2_agent.JolokiaAgent{
```

Comment:
SetupPlugin was split into two versions, one here for agent, and another in jolokia2_proxy/jolokia2_proxy_test.go. The proxy version checks the name before making the plugin. It's probably worth doing the same here.

```suggestion
		if name == "jolokia2_agent" {
			plugin := jolokia2_agent.JolokiaAgent{
```